src/radon.ts

```diff
@@ -1,423 +1,527 @@
+function getArgumentInfoType(info: ArgumentInfo): MarkupArgumentType {
+  if (info.type === MirArgumentType.FilterFunction) {
+    return MarkupArgumentType.SelectFilter
+  } else if (info.type === MirArgumentType.ReducerFunction) {
+    return MarkupArgumentType.SelectReduce
+  } else {
+    return MarkupArgumentType.Input
   }
 }
 
-// TODO: Call this function just at the beginning
-function generateFilterArgumentOptions(): Array<MarkupOption> {
+export function generateFilterArgumentOptions(): Array<MarkupOption> {
   const markupOptions: Array<MarkupOption> = getEnumNames(Filter).map(name => {
     return {
       label: name,
       hierarchicalType: MarkupHierarchicalType.OperatorOption,
       markupType: MarkupType.Option,
       // TODO: Add support for pseudotypes
-      outputType: OutputType.Bytes,
+      outputType: OutputType.FilterOutput,
     }
   })
   return markupOptions
 }
 
-// TODO: Call this function just at the beginning
-function generateReducerArgumentOptions(): Array<MarkupOption> {
+export function generateReducerArgumentOptions(): Array<MarkupOption> {
   const markupOptions: Array<MarkupOption> = getEnumNames(Reducer).map(name => {
     return {
       label: name,
       hierarchicalType: MarkupHierarchicalType.OperatorOption,
       markupType: MarkupType.Option,
-      outputType: OutputType.Bytes,
+      outputType: OutputType.ReducerOutput,
     }
   })
   return markupOptions
 }
+
+function getMirOperatorInfo(
+  operator: MirOperator
+): { code: OperatorCode; args: Array<MirArgument> } {
+  return Array.isArray(operator)
+    ? {
+        code: operator[0] as OperatorCode,
+        args: operator.slice(1) as Array<MirArgument>,
+      }
+    : {
+        code: operator as OperatorCode,
+        args: [],
+      }
+}
+
+function getDefaultMirArgumentByType(type: MirArgumentType): MirArgument {
+  switch (type) {
+    case MirArgumentType.Boolean:
+      return true
+    case MirArgumentType.FilterFunction:
+      return [Filter.LessThan, 0]
+    case MirArgumentType.Float:
+      return 0.0
+    case MirArgumentType.Integer:
+      return 0
+    case MirArgumentType.ReducerFunction:
+      return Reducer.averageMean
+    case MirArgumentType.String:
+      return ''
+    case MirArgumentType.Subscript:
+      return ''
+  }
+}
+
+// TODO: Refactor to find the first operator instead of repeat code
+function getDefaultMirOperatorByType(type: Type): MirOperator {
+  switch (type) {
+    case Type.Array:
+      return OperatorCode.ArrayCount
+    case Type.Boolean:
+      return [OperatorCode.BooleanMatch, '', true]
+    case Type.Bytes:
+      return OperatorCode.BytesAsString
+    case Type.Float:
+      return OperatorCode.FloatAbsolute
+    case Type.Integer:
+      return OperatorCode.IntegerAbsolute
+    case Type.Map:
+      return OperatorCode.MapEntries
+    case Type.String:
+      return OperatorCode.StringAsBoolean
+  }
+}
+
+function isArrayType(type: OutputType) {
+  return (
```

Comment:
In python it's idiomatic to do

    [OutputType.Array, OutputType.ArrayArray, ...].contains(type)

Reply:
But in javascript it's probably more idiomatic to just

    type.toString().startsWith("Array")